src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -0,0 +1,159 @@
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => new ThaiBuddhistCalendar().ToDateTime(minDate.Year + yearOffset - 1, maxDate.Month, maxDate.Day, maxDate.Hour, maxDate.Minute, maxDate.Second, maxDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year + yearOffset + 1, maxDate.Month, maxDate.Day, maxDate.Hour, maxDate.Minute, maxDate.Second, maxDate.Millisecond, 1));
+
+            // Month is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(minDate.Year, minDate.Month - 1, minDate.Day, minDate.Hour, minDate.Minute, minDate.Second, minDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year, maxDate.Month + 1, maxDate.Day, maxDate.Hour, maxDate.Minute, maxDate.Second, maxDate.Millisecond, 1));
+
+            // Day is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(minDate.Year, minDate.Month, minDate.Day - 1, minDate.Hour, minDate.Minute, minDate.Second, minDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year, maxDate.Month, maxDate.Day + 1, maxDate.Hour, maxDate.Minute, maxDate.Second, maxDate.Millisecond, 1));
+
+            // Hour is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(minDate.Year, minDate.Month, minDate.Day, minDate.Hour - 1, minDate.Minute, minDate.Second, minDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year, maxDate.Month, maxDate.Day, maxDate.Hour + 1, maxDate.Minute, maxDate.Second, maxDate.Millisecond, 1));
+
+            // Minute is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(minDate.Year, minDate.Month, minDate.Day, minDate.Hour, minDate.Minute - 1, minDate.Second, minDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year, maxDate.Month, maxDate.Day, maxDate.Hour, maxDate.Minute + 1, maxDate.Second, maxDate.Millisecond, 1));
+
+            // Second is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(minDate.Year, minDate.Month, minDate.Day, minDate.Hour, minDate.Minute, minDate.Second - 1, minDate.Millisecond, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new ThaiBuddhistCalendar().ToDateTime(maxDate.Year, maxDate.Month, maxDate.Day, maxDate.Hour, maxDate.Minute, maxDate.Second + 1, maxDate.Millisecond, 1));
```

Comment:
These tests don't seem to be testing the same thing the original was.  These tests are good as well, but for example, the original test was validating that any value less than 0 or greater than 59 for seconds was invalid.  This test is validating that when you go below the min datetime by one second, it's invalid... that's different.  For example, if the min datetime were January 15th 1980 at 1:23:45, your test validates that going to a datetime for January 15th 1980 at 1:23:44 fails, but we also need to validate that trying to create January 15th 1980 at 1:24:61 fails and 1:24:-1 fails, etc.